beprof/curve.py

```diff
@@ -31,8 +31,12 @@ class Curve(np.ndarray):
     Has additional field - info.
     """
 
-    def __new__(cls, input_array):
+    def __new__(cls, input_array, **meta):
         obj = np.asarray(input_array).view(cls).copy()
+        if meta is None:
+            obj.metadata = {}
+        else:
+            obj.metadata = meta.copy()
```

Comment:
I see you used shallow copy. Do you think it is better here than deep copy (https://docs.python.org/3.5/library/copy.html) ?